packages/supabase_flutter/README.md

```diff
@@ -427,6 +427,160 @@ Supabase.initialize(
+          return;
+        }
+        await sharedPreferencesLocalStorage
+            .persistSession(jsonEncode(session.toJson()));
+        await hiveLocalStorage.removePersistedSession();
+      }
+      if (Hive.box(_hiveBoxName).isEmpty) {
+        final boxPath = Hive.box(_hiveBoxName).path;
+        await Hive.deleteBoxFromDisk(_hiveBoxName);
+
+        //Delete `auth` folder if it's empty
+        if (!kIsWeb && boxPath != null) {
+          final boxDir = File(boxPath).parent;
+          final dirIsEmpty = await boxDir.list().length == 0;
+          if (dirIsEmpty) {
+            await boxDir.delete();
+          }
+        }
+      }
+    }
+  }
+
+  @override
+  Future<String?> accessToken() {
+    return sharedPreferencesLocalStorage.accessToken();
+  }
+
+  @override
+  Future<bool> hasAccessToken() {
+    return sharedPreferencesLocalStorage.hasAccessToken();
+  }
+
+  @override
+  Future<void> persistSession(String persistSessionString) {
+    return sharedPreferencesLocalStorage.persistSession(persistSessionString);
+  }
+
+  @override
+  Future<void> removePersistedSession() {
+    return sharedPreferencesLocalStorage.removePersistedSession();
+  }
+}
+
+/// A [LocalStorage] implementation that implements Hive as the
+/// storage method.
+class HiveLocalStorage extends LocalStorage {
+  /// Creates a LocalStorage instance that implements the Hive Database
+  const HiveLocalStorage();
+
+  /// The encryption key used by Hive. If null, the box is not encrypted
+  ///
+  /// This value should not be redefined in runtime, otherwise the user may
+  /// not be fetched correctly
+  ///
+  /// See also:
+  ///
+  ///   * <https://docs.hivedb.dev/#/advanced/encrypted_box?id=encrypted-box>
+  static String? encryptionKey;
+
+  @override
+  Future<void> initialize() async {
+    HiveCipher? encryptionCipher;
+    if (encryptionKey != null) {
+      encryptionCipher = HiveAesCipher(base64Url.decode(encryptionKey!));
+    }
+    await Hive.initFlutter('auth');
+    await Hive.openBox(_hiveBoxName, encryptionCipher: encryptionCipher)
+        .timeout(const Duration(seconds: 1));
+  }
+
+  @override
+  Future<bool> hasAccessToken() {
+    return Future.value(
+      Hive.box(_hiveBoxName).containsKey(
+        supabasePersistSessionKey,
+      ),
+    );
+  }
+
+  @override
+  Future<String?> accessToken() {
+    return Future.value(
+      Hive.box(_hiveBoxName).get(supabasePersistSessionKey) as String?,
+    );
+  }
+
+  @override
+  Future<void> removePersistedSession() {
+    return Hive.box(_hiveBoxName).delete(supabasePersistSessionKey);
+  }
+
+  @override
+  Future<void> persistSession(String persistSessionString) {
+    // Flush after X amount of writes
+    return Hive.box(_hiveBoxName)
+        .put(supabasePersistSessionKey, persistSessionString);
+  }
+}
+```
+
+You can then initialize Supabase with `MigrationLocalStorage` and it will automatically migrate the sessino from Hive to SharedPreferences.
```

Comment:
```suggestion
You can then initialize Supabase with `MigrationLocalStorage` and it will automatically migrate the session from Hive to SharedPreferences.
```